### PR TITLE
added Path to IDirectory and IFile

### PR DESCRIPTION
**Pull request by DanielWare**
I think we will need to expose the Path for both Directory and File. While testing out some implementation with dotnet 5, specifically the CopyTo Function. We will need a way to get the path of the passed in IFile destination.

### CODE REVIEW

File: FileSystem.Fluent.Abstractions.Net5/Net5Directory.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        public IDirectory TryCreate()
+        {
+            try
+            {
+                Directory.CreateDirectory(Path);
+                return this;
+            }
+            catch (Exception e)
+            {
```
Comment (michaelkoss) [inline FileSystem.Fluent.Abstractions.Net5/Net5Directory.cs]: Let’s not write line, we should probably just throw the exception and let the clients handle it.

File: FileSystem.Fluent.Abstractions.Net5/Net5File.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        public IFile OpenRead(Action<Stream> action)
+        {
+            using Stream fileStream = new FileStream(Path, FileMode.OpenOrCreate, FileAccess.Read);
+            action?.Invoke(fileStream);
+            return this;
+        }
+
+        public IFile WriteContents(string contents, Encoding encoding = null)
+        {
+            byte[] bytes = Encoding.Default.GetBytes(contents);
```
Comment (michaelkoss) [inline FileSystem.Fluent.Abstractions.Net5/Net5File.cs]: If encoding is passed, we should use it, otherwise Encoding.Default

Comment (michaelkoss) [reply FileSystem.Fluent.Abstractions.Net5/Net5File.cs]: Same comment in a few places

**Review by michaelkoss — CHANGES_REQUESTED** [review]

**Comment by DanielWare** [comment]
Oh I didn't mean for my initial dotnet 5 implementation to make it to this PR. I'll have to do some digging into how fork PRs work. Good changes though. I'll make those.

**Comment by michaelkoss** [comment]
We talked about adding a property for ‘FileInfo’ to get the properties of the file. Should ‘Path’ go in there?

**Comment by DanielWare** [comment]
That could be a good idea. I would assume that we would want all the properties in that object to be lazy loaded or maybe even never persisted (ie we get modified date from the OS every time it's requested)? Path would be an exception, since it wouldn't change.

**Comment by michaelkoss** [comment]
I'm not sure how we would want to handle this. If we asked for the file
info, then ask for many properties off that file info, we wouldn't want a
call to the OS for each property on the file info. But, we would want that
to time out pretty quickly. Thoughts?

**Pull request closed without merging**